enterprise_access/apps/api/v1/views.py

```diff
@@ -1 +1,163 @@
+
+        if not customer_configuration.subsidy_type:
+            error_msg = f'Subsidy request type for enterprise: {enterprise_customer_uuid} has not been set up.'
+            raise SubsidyRequestCreationError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+        if customer_configuration.subsidy_type != self.subsidy_type:
+            error_msg = f'Subsidy request type must be {customer_configuration.subsidy_type}'
+            raise SubsidyRequestCreationError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+    @permission_required(
+        constants.REQUESTS_ADMIN_LEARNER_ACCESS_PERMISSION,
+        fn=get_enterprise_uuid_from_request_data,
+    )
+    def create(self, request, *args, **kwargs):
+        try:
+            self._validate_subsidy_request()
+        except SubsidyRequestCreationError as exc:
+            logger.exception(exc)
+            return Response(exc.message, exc.http_status_code)
+
+        # Set the lms user id for the request
+        request.data['lms_user_id'] = self.lms_user_id
+        return super().create(request, *args, **kwargs)
+
+class LicenseRequestViewSet(SubsidyRequestViewSet):
+    """ Viewset for license requests. """
+
+    queryset = LicenseRequest.objects.order_by('-created')
+    serializer_class = serializers.LicenseRequestSerializer
+
+    subsidy_type = SubsidyTypeChoices.LICENSE
+
+    def _validate_subsidy_request(self):
+        super()._validate_subsidy_request()
+
+        enterprise_customer_uuid = self.request.data.get('enterprise_customer_uuid')
+
+        pending_request = LicenseRequest.objects.filter(
+            lms_user_id=self.lms_user_id,
+            enterprise_customer_uuid=enterprise_customer_uuid,
+            state__in=[SubsidyRequestStates.PENDING_REVIEW, SubsidyRequestStates.APPROVED_PENDING]
+        ).first()
+
+        if pending_request:
+            error_msg = f'User already has an outstanding license request for enterprise: {enterprise_customer_uuid}.'
+            logger.exception(error_msg)
+            raise SubsidyRequestCreationError(error_msg, status.HTTP_422_UNPROCESSABLE_ENTITY)
+
+class CouponCodeRequestViewSet(SubsidyRequestViewSet):
+    """ Viewset for coupon code requests. """
+
+    queryset = CouponCodeRequest.objects.order_by('-created')
+    serializer_class = serializers.CouponCodeRequestSerializer
+
+    subsidy_type = SubsidyTypeChoices.COUPON
+
+    def _validate_subsidy_request(self):
+        super()._validate_subsidy_request()
+
+        enterprise_customer_uuid = self.request.data.get('enterprise_customer_uuid')
+        course_id = self.request.data.get('course_id')
+
+        pending_request = CouponCodeRequest.objects.filter(
+            lms_user_id=self.lms_user_id,
+            enterprise_customer_uuid=enterprise_customer_uuid,
+            state__in=[SubsidyRequestStates.PENDING_REVIEW, SubsidyRequestStates.APPROVED_PENDING],
+            course_id=course_id
+
+        ).first()
```

Comment:
whenever I see `filter().first()` i get worried, usually because we never know what that thing is that we're getting back because #ordering. if you just want to check something is returned, i would use `.exists()`